model/variable.cc

```diff
@@ -204,8 +204,12 @@ string VariableBase::_init(const string& x)
               (type()==flow && !cminsky().definingVar(valueId()))) 
             val.reset(minsky().variableValues);
         }
-      catch (...)
-        {}
+      // Catch exception "Unknown variable XX in initialisation of YY" at input time. For ticket 1045.
```

Comment:
I think we do want this error message to propagate to the user, which is why I think this fix doesn't work.

Reply:
I also think it is better for the exception to be caught on variableValue reset, which, as you allude to in the same member, is related to an older ticket (#433). I guess the bug becomes less important, because as a result of some posterior fix, it is now possible to edit the variable whose value disappeared permanently and was previously uneditable. But it doesn't look good. So, would a requestRedraw() fix the disappearing values on the other variables?